instrumentation/opentelemetry-instrumentation-tornado/src/opentelemetry/instrumentation/tornado/__init__.py

```diff
@@ -255,11 +256,13 @@ def _get_operation_name(handler, request):
 def _start_span(tracer, handler, start_time) -> _TraceContext:
     token = context.attach(extract(handler.request.headers))
 
-    span = tracer.start_span(
+    server_span_tracer = ServerSpanTracer(
```

Comment:
This API means we extract the context every time irrespective of whether a parent span is present or not. I know it might be relatively cheap but for a service that can serve thousands of requests a second, unnecessary attribute lookups and garbage generation can quickly add up to affect performance. 

I'd change so that we first look for current span and only extract from carrier if we don't find one. 

Also, this creates one instance of ServerSpanTracer per request. Otel API was specifically designed in a way to minimize allocations by re-using things like tracers, context getters/setters, etc. We should follow that pattern and not generate new objects unnecessarily.

Reply:
That makes sense; the only reason I created the ServerSpanTracer API was so that I would be able to reuse the logic of checking for a span in the current context just by calling a single `start_span()` method. I think it might be easier just to get rid of this package and API and instead use the logic directly in all the instrumentations that create server spans. If I keep using the API, I will have to add more logic for only conditionally extracting context.

Reply:
makes sense to me. you can try having some helper functions to reuse code a bit if possible but not a requirement.